sycl/doc/extensions/proposed/sycl_ext_oneapi_graph_fusion.asciidoc

```diff
@@ -0,0 +1,777 @@
+        range{dataSize},
+        {sycl::ext::oneapi::experimental::property::access_scope_work_item{},
+          sycl::ext::oneapi::experimental::property::fusion_internal_memory{},
+          no_init}};
+    // Internalization specified on the buffer
+    buffer<int> bTmp3{
+        range{dataSize},
+        {sycl::ext::oneapi::experimental::property::access_scope_work_item{},
+          sycl::ext::oneapi::experimental::property::fusion_internal_memory{},
+          no_init}};
+    buffer<int> bOut{out, range{dataSize}};
+    bOut.set_write_back(false);
+
+    ext::oneapi::experimental::command_graph graph{
+        q.get_context(), q.get_device(),
+        sycl::ext::oneapi::experimental::property::graph::
+                                                assume_buffer_outlives_graph{}};
+
+    graph.begin_recording(q);
+
+    q.submit([&](handler &cgh) {
+      auto accIn1 = bIn1.get_access(cgh);
+      auto accIn2 = bIn2.get_access(cgh);
+      // Internalization specified on each accessor.
+      auto accTmp1 = bTmp1.get_access(cgh,
+            sycl::ext::oneapi::experimental::property::access_scope_work_item{}
+            sycl::ext::oneapi::experimental::property::fusion_internal_memory{},
+            no_init);
+      cgh.parallel_for<AddKernel>(dataSize, AddKernel{accIn1, accIn2, accTmp1});
+    });
+
+    q.submit([&](handler &cgh) {
+      // Internalization specified on each accessor.
+      auto accTmp1 = bTmp1.get_access(cgh,
+            sycl::ext::oneapi::experimental::property::access_scope_work_item{}
+            sycl::ext::oneapi::experimental::property::fusion_internal_memory{},
+            no_init);
+      auto accIn3 = bIn3.get_access(cgh);
+      auto accTmp2 = bTmp2.get_access(cgh);
+      cgh.parallel_for<class KernelOne>(
+          dataSize, [=](id<1> i) { accTmp2[i] = accTmp1[i] * accIn3[i]; });
+    });
+
+    q.submit([&](handler &cgh) {
+      // Internalization specified on each accessor.
+      auto accTmp1 = bTmp1.get_access(cgh,
+            sycl::ext::oneapi::experimental::property::access_scope_work_item{}
+            sycl::ext::oneapi::experimental::property::fusion_internal_memory{},
+            no_init);
+      auto accTmp3 = bTmp3.get_access(cgh);
+      cgh.parallel_for<class KernelTwo>(
+          dataSize, [=](id<1> i) { accTmp3[i] = accTmp1[i] * 5; });
+    });
+
+    q.submit([&](handler &cgh) {
+      auto accTmp2 = bTmp2.get_access(cgh);
+      auto accTmp3 = bTmp3.get_access(cgh);
+      auto accOut = bOut.get_access(cgh);
+      cgh.parallel_for<AddKernel>(dataSize,
+                                  AddKernel{accTmp2, accTmp3, accOut});
+    });
+
+    graph.end_recording();
+
+    // Trigger fusion during finalization.
+    auto exec_graph =
+        graph.finalize({sycl::ext::oneapi::experimental::property::
+                            graph::require_fusion{}});
+
+    q.ext_oneapi_graph(exec_graph);
+
+    q.wait();
+  }
+  return 0;
+}
+```
+
+=== USM-based example
```

Comment:
NIT: Can we modify this example so we use the `access_scope_work_group` property instead?

Reply:
Done.